kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/RecordConverter.java

```diff
@@ -126,10 +127,14 @@ private Object convertValue(
       case STRING:
         return convertString(value);
       case UUID:
+        if ("parquet".equals(config.writeProps().get("write.format.default"))) {
+          return UUIDUtil.convert(convertUUID(value));
+        }
         return convertUUID(value);
       case BINARY:
-      case FIXED:
         return convertBase64Binary(value);
+      case FIXED:
+        return convertBase64Binary(value).array();
```

Comment:
ByteBuffers.toByteArray(...)

Also just making  sure, we won't be re-using this buffer anywhere?

Reply:
+1 for `ByteBuffers.toByteArray()`. The buffer shouldn't be reused, a record is converted to an Iceberg row, written, then dereferenced. I don't believe any of the writers cache rows as-is.